packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2024, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  forwardRef,
+  useCallback,
+  useEffect,
+  useId,
+  useRef,
+  useState,
+  type ChangeEventHandler,
+} from 'react';
+
+import { Input, type InputElement, type InputProps } from '../Input/index.js';
+import { clsx } from '../../styles/clsx.js';
+import { FieldLabel, FieldLabelText } from '../Field/index.js';
+import { applyMultipleRefs } from '../../util/refs.js';
+
+import styles from './ColorInput.module.css';
+
+export interface ColorInputProps
+  extends Omit<
+    InputProps,
+    | 'ref'
+    | 'type'
+    | 'defaultValue'
+    | 'value'
+    | 'placeholder'
+    | 'maxLength'
+    | 'pattern'
+    | 'renderPrefix'
+    | 'renderSuffix'
+    | 'as'
+  > {
+  /**
+   * A short string that is shown inside the empty input.
+   */
+  placeholder?: string;
+  /**
+   * The value of the input element.
+   */
+  value?: string;
+  /**
+   * The default value of the input element.
+   */
+  defaultValue?: string;
+  /*
+   * Picker label describes the color picker which serves as an alternative
+   * to the hex color input.
+   */
+  pickerLabel: string;
+}
+
+export const ColorInput = forwardRef<InputElement, ColorInputProps>(
+  (
+    { onChange, className, value, defaultValue, pickerLabel, ...props },
+    ref,
+  ) => {
+    const [currentColor, setCurrentColor] = useState<string | undefined>(
+      defaultValue,
+    );
+    const colorDisplayRef = useRef<HTMLDivElement>(null);
+    const colorPickerRef = useRef<InputElement>(null);
+    const pickerId = useId();
+
+    const onPickerColorChange: ChangeEventHandler<InputElement> = (e) => {
+      setCurrentColor(e.target.value);
+      if (onChange) {
+        onChange(e);
+      }
+    };
+
+    const onInputChange: ChangeEventHandler<InputElement> = (e) => {
+      if (colorPickerRef.current) {
+        colorPickerRef.current.value = `#${e.target.value}`;
+      }
+      setCurrentColor(`#${e.target.value}`);
+    };
+
+    useEffect(() => {
+      if (colorDisplayRef.current && currentColor) {
+        colorDisplayRef.current.style.backgroundColor = currentColor;
+      }
+    }, [currentColor]);
```

Comment:
Took me a while to recall but now I remember 😅 so - `renderSuffix` is created only once on the first render with initial color. On any subsequent color change the `useEffect` is used instead. This was necessary because what kept happening otherwise was that as a user you clicked on some color from the native color picker, `currentColor` changed, and because the input field "didn't exist anymore" the browser dismissed the native color picker widget.

Reply:
Ugh, React's re-render hell strikes again 😞 

I think we can find a better approach though. Wanna pair on this?

Reply:
Yes please 👍

Reply:
Resolved.